Raven.Database/Config/CpuStatistics.cs

```diff
@@ -98,10 +118,35 @@ private static void HandleCpuUsage(float usageInPercents)
             if (average < 0)
                 return; // there was an error in getting the CPU stats, ignoring
 
+            var enumerator = cpuUsageCallsRecords.GetEnumerator();
+
             if (average >= HighNotificationThreshold)
+            {
+
+                if (!enumerator.Current.Reason.Equals("High CPU usage"))
```

Comment:
This should throw. You didn't call the `MoveNext` first.
Also, if you want to get the last in the list, you better use Peek(), instead.
And comparing string is bad, if you care for the value, set a property there.